num/src/error.rs

```diff
@@ -37,6 +37,21 @@ impl core::fmt::Display for OverflowError {
 #[cfg(feature = "std")]
 impl std::error::Error for OverflowError {}
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum DivError {
+    ZeroDiv,
+    Overflow,
+}
+
+impl core::fmt::Display for DivError {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        match *self {
+            DivError::ZeroDiv => write!(f, "division by zero"),
+            DivError::Overflow => write!(f, "division with overflow"),
```

Comment:
We normally use `#[derive(Display)]` for that, providing the struct with doc comments and `#[display(doc_comments)]`